reststops.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+Jeffrey Chen
+35 minutes
+10/10 test cases
+4/10 difficulty
+It was pretty easy for a silver problem, I just needed to remember to use long long instead of int.
+*/
+
+#include <algorithm>
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <vector>
+#include <queue>
+#include <stack>
+#include <map>
+#include <set>
+#include <unordered_map>
+#include <unordered_set>
+
+using namespace std;
+
+int main(int argc, char* argv[]) {
+	const char* FIN = "reststops.in";
+	const char* FOUT = "reststops.out";
+	const char* inFile = argc > 1 ? argv[1] : FIN;
+	ifstream cin(inFile);
+	ofstream cout(FOUT);
+
+	long long l, n, rf, rb;
+	cin >> l >> n >> rf >> rb;
+
+	vector<pair<long long, long long>> stops;
+	for (long long i = 0; i < n; ++i) {
+		pair<long long, long long> temp;

```

Comment:
The problem with `pair` is you always need recall the meaning of "first" and "second", these are frictions. If you are a serious competitor, you remove all frictions.